### DISCUSSION TITLE
1-wire temperature sensor doesn't update when no water flow

### DISCUSSION THREAD

**Discussion by jslabotsky** [tagyoureit/nodejs-poolController#678] [Nixie]
Okay I had originally written a very long winded post and then realized I completely misinterpreted what was going on. So if you saw that, forget everything I said.

I just upgraded to njsPC 7.7 and the latest Dashpanel and REM. The water temperature sensor does not update when no pump circuit is active. Is there a way to change this? I would like to keep tabs on the temperature of the water in the pipes when the pump is off.

**Comment by tagyoureit** [comment]
I did read your long post.  :)  Answer is the same.

REM will continue to read the temps and even send them to njsPC.  However, njsPC will ignore them unless the pool (or spa) circuit is on.  This is by design.

**Comment by jslabotsky** [comment]
What is the rationale for ignoring the temp when the pool circuit is off? It didn't do that before. The dashpanel continues to display the last temperature that was received, albeit grayed out, so it is a bit misleading. I did figure out there is a different MQTT message I can read in Home Assistant to get the temperature all the time, so in the end I was able to makeit meet my needs. I just wasnt prepared for this behavior to change when I upgraded. I guess one could also create a circuit with the pump set at zero RPM so that the pool circuit is always on.
    On Saturday, January 7, 2023 at 10:21:09 AM CST, tagyoureit ***@***.***> wrote:  
 
 


I did read your long post. :) Answer is the same.

REM will continue to read the temps and even send them to njsPC. However, njsPC will ignore them unless the pool (or spa) circuit is on. This is by design.

**Reply by tagyoureit** [reply]
It is this way because a small amount of water that is sitting in your pipes and becomes hotter (or colder) than your pool isn't representative of the larger body of water.  There are also complications with shared bodies where the same sensor is feeding both the pool and spa temps.  You wouldn't want to be reporting that your pool is 100 degrees when you have a spa one.  

I'll ask your question back to you... why does it matter what the temperature of the water in your pipes is? 

And, yes, we'll make changes and not everything gets documented.  But most of the time they are good changes.  ;-)

**Comment by jslabotsky** [comment]
It matters because I don't shut down my pool in the winter. I run my pump when the temperature falls below freezing. I have had an occasion when a the ambient temperature rose rapidly above the freeze protection threshold (40F) but water in the pipes remained near freezing. It resulted in ice forming that did not burst the pipes on its own but did form a significant blockage. When the pump turned back on as the temperature fell below the freeze protection threshold, priming created excessive pressure and burst the pipes in spectacular fashion. It was a very unusual circumstance, but reading the temperature of the static water serves as a backstop in case something odd like that happens again. I have moved the freeze protection automation into Home Assistant and added some safety measures that should render this moot, but at this point I'm looking for every bit of reassurance I can get.
I wouldn't call it a *bad* change, but it was one that I had not accounted for and it became very consequential for me. Since I can get the information I want from REM it's moot, so I'm certainly not asking you to change anything at this point.
    On Saturday, January 7, 2023 at 07:15:14 PM CST, tagyoureit ***@***.***> wrote:  
 
 


It is this way because a small amount of water that is sitting in your pipes and becomes hotter (or colder) than your pool isn't representative of the larger body of water. There are also complications with shared bodies where the same sensor is feeding both the pool and spa temps. You wouldn't want to be reporting that your pool is 100 degrees when you have a spa one.

I'll ask your question back to you... why does it matter what the temperature of the water in your pipes is?

And, yes, we'll make changes and not everything gets documented. But most of the time they are good changes. ;-)

**Reply by tagyoureit** [reply]
First off, that sucks.  :(

The way freeze protection works is that it is based on the ambient temp.  When the temp falls near/below freezing the pumps should always be running.  In a shared body the valves alternate between the pool/spa.  So long as the water is running then it will not build up ice anywhere in the pumps.  What temperature does your pool get down to in winter?  It's surprising to hear that the ambient temp went up to 40 but ice still formed in your pipes.

**Comment by jslabotsky** [comment]
Eh, it gave me a chance to  make some plumbing refinements. I live in Oklahoma and we have some pretty wild temperature swings.But typically winters are relatively mild so the pool itself generally stays above freezing. The main concern is keeping the equipment thawed out. However, this year and the winter of 20-21 had some blips of more extreme weather we're not accustomed to.
When the "bomb cyclone" dropped into the U.S. around Christmas, our temps dropped from the 50's down to about 15F over the course of a few hours and stayed there for about a day and a half. Normally that's no big deal, but this time it came with two days of non-stop 30-35+ MPH wind. My pool has no wind break whatsoever (which I have plans to change) It was blowing along the length of my pool so hard that water was splashing out on the deck until the surface finally froze.
On the third day a warm front swept in and the sun came out, so ambient temp went up into the low 40's about as fast as it had fallen. The Home Assistant logs indicated everyhing was running normally until freeze protection turned off. I noticed the problem when I checked on the system after the sun got low and could see it had failed to prime. That's when I began to investigate. I'm guessing there was some amount of ice buildup before the pump shut off and it just solidified when water stopped flowing. It should also be noted the equipment is in the shade most of the day, so the sun didn't help warm it up.
Contrast this with Feb. 2021 when I made it through temps as low as -15F for a few days and survived. I did take a few extra steps. I put tarps over the equipment pad and stuck a couple of halogen work lights in there to warm the equipment a little bit. (Old trick I've used to thaw frozen pipes in a crawl space.) And because it was so extremely cold I ran the pump at a higher speed. I just didn't think all fhat would be necessary this time around. Lesson learned I suppose. And don't get me wrong - if I'd had any reason to think it would get down to -15 I definitely would have winterized the pool instead. I'm 50, lived here my whole life, and I've never seen it get that cold here. 
 Next year I'll probably give in and winterize, but this winter I'm already in it pretty deep, so no sense giving up now,

    On Saturday, January 7, 2023 at 08:48:53 PM CST, tagyoureit ***@***.***> wrote:  
 
 


First off, that sucks. :(

The way freeze protection works is that it is based on the ambient temp. When the temp falls near/below freezing the pumps should always be running. In a shared body the valves alternate between the pool/spa. So long as the water is running then it will not build up ice anywhere in the pumps. What temperature does your pool get down to in winter? It's surprising to hear that the ambient temp went up to 40 but ice still formed in your pipes.
